### DISCUSSION TITLE
MidiFile.GetTrackChunks() VS Chunks.OfType<TrackChunk>()

### DISCUSSION THREAD

**Discussion by Jenix-Park** [melanchall/drywetmidi#185] [Q&A] [answered]
What is the difference between MidiFile.GetTrackChunks() and MidiFile.Chunks.OfType<TrackChunk>()?
Is there any performance difference?

**Comment by melanchall** [comment, marked as answer]
Those two things are absolutely the same, there is no any difference in performance. `MidiFile.GetTrackChunks()` is just a shortcut for `MidiFile.Chunks.OfType<TrackChunk>()`.

**Reply by Jenix-Park** [reply]
Oh, I see.

Then does the same apply to the following too?

```
1)
using (var notesManager = trackChunk.ManageNotes())
	foreach (var notes in notesManager.Objects) { }

2)
foreach (var notes in trackChunk.Events.OfType<Note>()) { }

3)
using (var notesManager = new TimedObjectsManager<Note>(trackChunk.Events))
	foreach (var notes in notesManager.Objects) { }
```

**Reply by melanchall** [reply]
1 and 3 are the same things. 2 is wrong at all. `trackChunk.Events` returns `EventsCollection` which holds collection of `MidiEvent`, which cannot be casted to `Note`. The library has full API documentation, please see [here about Events property](https://melanchall.github.io/drywetmidi/api/Melanchall.DryWetMidi.Core.TrackChunk.html#Melanchall_DryWetMidi_Core_TrackChunk_Events).

**Reply by Jenix-Park** [reply]
Oops..
I read them all when I first used your great DryWetMIDI plugin,
but it's a very long time ago, so I forgot many things and couldn't even fully understand what I wrote before. lol
The number two was in fact 'var setTempoEvent in trackChunk.Events.OfType<SetTempoEvent>()', and I thought it would work with Note too.
Thank you, as always :)
